symplyphysics/laws/optics/irradiance_of_light_after_polarizer.py

```diff
@@ -0,0 +1,57 @@
+from sympy import Eq, solve, cos
+from symplyphysics import (
+    units,
+    Quantity,
+    Symbol,
+    print_expression,
+    validate_input,
+    validate_output,
+    dimensionless,
+    angle_type,
+)
+
+# Description
+## Malus's law states that the irradiance of linearly polarized light that passes through a polarizer
+## is proportional to its initial intensity I0, transparency coefficient k of the polarizer
+## and square cosine of the angle phi between the light's initial polarization direction and the axis
+## of the polarizer
+
+# Law: I = I0 * k * cos^2(phi)
+## I is the irradiance of light passed through the polarizer
+## I0 is the initial irradiance
+## k is the polarizer's transparency coefficient
+## phi is the angle between the initial polarization direction and the axis of the polarizer
+
+irradiance_final = Symbol("irradiance_final", units.watt / units.meter**2)
+irradiance_initial = Symbol("irradiance_initial", units.watt / units.meter**2)
+transparency_coefficient = Symbol("transparency_coefficient", dimensionless)
+polarization_angle = Symbol("polarization_angle", angle_type)
+
+law = Eq(
+    irradiance_final,
+    irradiance_initial * transparency_coefficient * cos(polarization_angle) ** 2,
+)
+
+
+def print_law() -> str:
+    return print_expression(law)
+
+
+@validate_input(
+    irradiance_initial_=irradiance_initial,
+    transparency_coefficient_=transparency_coefficient,
+    polarization_angle_=polarization_angle,
+)
+@validate_output(irradiance_final)
+def calculate_irradiance(
+    irradiance_initial_: Quantity,
+    transparency_coefficient_: Quantity,
+    polarization_angle_: Quantity,
```

Comment:
In other laws we have an option of passing angle as float. You might like to do it as well. See an example: https://github.com/blackyblack/symplyphysics/blob/17d00b153fc87168dba61bdf9eb94ca53dd2d14f/symplyphysics/laws/kinematic/planar_projection_is_cosine.py#L36